ui/webapp.py

```diff
@@ -12,175 +13,197 @@
 import SessionState
 from utils import feedback_doc, haystack_is_ready, retrieve_doc, upload_doc
 
+
 # Adjust to a question that you would like users to see in the search bar when they load the UI:
 DEFAULT_QUESTION_AT_STARTUP = "Who is the father of Arya Stark?"
 
+# Labels for the evaluation
+EVAL_LABELS = os.getenv("EVAL_FILE", Path(__file__).parent / "eval_labels_example.csv")
 
-def annotate_answer(answer, context):
-    """ If we are using an extractive QA pipeline, we'll get answers
-    from the API that we highlight in the given context"""
-    start_idx = context.find(answer)
-    end_idx = start_idx + len(answer)
-    # calculate dynamic height depending on context length
-    height = int(len(context) * 0.50) + 5
-    annotated_text(context[:start_idx], (answer, "ANSWER", "#8ef"), context[end_idx:], height=height)
-
-
-def show_plain_documents(text):
-    """ If we are using a plain document search pipeline, i.e. only retriever, we'll get plain documents
-    from the API that we just show without any highlighting"""
-    st.markdown(text)
-
-
-def random_questions(df):
-    """
-    Helper to get one random question + gold random_answer from the user's CSV 'eval_labels_example'.
-    This can then be shown in the UI when the evaluation mode is selected. Users can easily give feedback on the
-    model's results and "enrich" the eval dataset with more acceptable labels
-    """
-    random_row = df.sample(1)
-    random_question = random_row["Question Text"].values[0]
-    random_answer = random_row["Answer"].values[0]
-    return random_question, random_answer
+# Whether the file upload should be enabled or not
+DISABLE_FILE_UPLOAD = os.getenv("HAYSTACK_UI_DISABLE_FILE_UPLOAD")
 
 
 def main():
-    # Define state
-    state_question = SessionState.get(
-        random_question=DEFAULT_QUESTION_AT_STARTUP, random_answer="", next_question="false", run_query="false"
+
+    # Persistent state
+    state = SessionState.get(
+        random_question=DEFAULT_QUESTION_AT_STARTUP, 
+        random_answer="",
+        results=None,
+        raw_json=None,
+        get_next_question=True
     )
 
-    # Initialize variables
-    eval_mode = False
-    random_question = DEFAULT_QUESTION_AT_STARTUP
-    eval_labels = os.getenv("EVAL_FILE", "eval_labels_example.csv")
+    # Small callback to reset the interface in case the text of the question changes
+    def reset_results(*args):
+        state.results = None
+        state.raw_json = None
+
+    # Title
+    st.write("# Countries of the World")
```

Comment:
I thought "Haystack Demo" (or "haystack Demo") is a good title. Maybe check with Andrey regarding SEO?

Reply:
Yep also this shouldn't have made it to this PR, sorry :sweat_smile: Rolling back :+1: